lib/configuration.js

```diff
@@ -195,13 +208,17 @@ confProto.openH2StreamForPush = function (pushUrl, proxyTransportUrl) {
     var openH2StreamForPush, reopenH2StreamForPush, // expression not function,
         request,
         reconnectFailures = [],
-        reconnectAutoDelay = self.options.reconnectAutoDelay || 100,
-        reconnectAuto = self.options.reconnectAuto || true,
+        options = self.options,
+        reconnectAutoDelay = options.reconnectAutoDelay || 100,
+        reconnectAuto = options.reconnectAuto || true,
```

Comment:
reconnectAuto will always be set to true with this logic because it does not make a distinction between `false` and `undefined` 

Assuming I understand the intent, the correct logic can be written:

```javascript
reconnectAuto = options.reconnectAuto === undefined || options.reconnectAuto,
```

Reply:
fixed via default options in constructor